sympy/solvers/ode/single.py

```diff
@@ -232,14 +236,17 @@ def _matches(self):
         if order != 1:
             return False
 
-        eq = expand(eq / eq.coeff(df))
-        eq = eq.collect(f(x), func = cancel)
-
         pattern = self._equation(f(x), x, 1)
 
+        if not isinstance(pattern.coeff(df), Wild):
+            eq = expand(eq / eq.coeff(df))
+        eq = eq.collect(f(x), func = cancel)
```

Comment:
Probably `if not pattern.coeff(df).has(Wild)` would be better in case it's something like `2*a` where `a` is a `Wild`